src/hooks/useMapSetup.ts

```diff
@@ -0,0 +1,273 @@
+import { getUrlParameter, setUrlParameter } from '../utilities/UrlParameters';
+import { useFilter } from './useFilter';
+import useRemoteConfigs from './useRemoteConfigs';
+
+type LayerNameKey = keyof LayerNames;
+
+export function useMapSetup(
+  mapNodeRef: React.RefObject<HTMLDivElement | null>,
+  trayIsOpen: boolean,
+  useMyLocationOnLoad: boolean,
+  isSmallScreen: boolean,
+  onFeatureIdentify: (graphic: __esri.Graphic | null) => void,
+  popupRoot: HTMLDivElement,
+  feedbackButtonRef: React.RefObject<HTMLDivElement | null>,
+  zoomButtonRef: React.RefObject<HTMLDivElement | null>,
+  setCenter: (center: number[]) => void,
+) {
+  const getConfig = useRemoteConfigs();
+  const map = useRef<WebMap | null>(null);
+  const mapView = useRef<MapView>(null);
+  const layers = useRef<Record<LayerNameKey, __esri.FeatureLayer | null>>({
+    routeTypes: null,
+    trafficStress: null,
+    trafficSignals: null,
+    otherLinks: null,
+  });
+  const mapIsInitialized = useRef<boolean>(false);
+  const highlightHandle = useRef<__esri.Handle>(null);
+  const { dispatch } = useFilter();
+
+  const logEvent = useFirebaseAnalytics();
+
+  // set up map
+  useEffect(() => {
+    if (!mapNodeRef.current || mapIsInitialized.current || !getConfig) {
+      return;
+    }
+
+    const initializeMap = async () => {
+      console.log('setting up the map');
+      mapIsInitialized.current = true;
+      map.current = new WebMap({
+        portalItem: {
+          id: getConfig('webMapId') as string,
+        },
+      });
+
+      let center: __esri.PointProperties | number[] = {
+        // SLC
+        x: -12455376,
+        y: 4978678,
+        spatialReference: {
+          wkid: 102100,
+        },
+      };
+
+      const xyUrlParam = getUrlParameter('center', 'number[]') as number[];
+      const zoomUrlParam = getUrlParameter('zoom', 'number') as number;
+      if (xyUrlParam) {
+        center = {
+          x: xyUrlParam[0],
+          y: xyUrlParam[1],
+          spatialReference: {
+            wkid: 3857,
+          },
+        };
+      } else if (useMyLocationOnLoad) {
+        const location = await getCoarseLocation();
+        if (location) {
+          center = [location.x, location.y];
+        }
+      }
+
+      const view = new MapView({
+        container: mapNodeRef.current,
+        center,
+        zoom: zoomUrlParam ?? INITIAL_MAP_ZOOM,
+        ui: {
+          components: isSmallScreen ? [] : ['zoom'],
+        },
+        map: map.current,
+        padding: {
+          bottom: trayIsOpen ? 320 : 0,
+        },
+        popup: {
+          actions: [],
+          dockEnabled: false,
+          dockOptions: {
+            breakpoint: false,
+            buttonEnabled: false,
+          },
+          visibleElements: {
+            actionBar: false,
+            heading: false,
+            closeButton: false,
+          },
+        },
+        popupEnabled: false,
+      });
+
+      view.on('immediate-click', (event) => {
```

Comment:
I've become annoyed with atlas after choosing immediate-click. It interferes with double click zoom, drag zooming, etc. I'd recommend changing this to a standard `click`.